libs/ui/src/components/navigator/navigator.tsx

```diff
@@ -88,16 +88,43 @@ export const Navigator: FC = () => {
   useResetLoading();
 
   useEffect(() => {
-    // TODO: Add check for ScreenEnum.AlmostDone screen later
-    const isCreateWalletScreensOpened =
+    const IsOpenOnlyFullViewScreen =
       initialState?.routes.some(
-        route => route.name === ScreensEnum.CreateANewWallet || route.name === ScreensEnum.VerifyMnemonic
+        route =>
+          route.name === ScreensEnum.AddNewCollectible ||
+          route.name === ScreensEnum.AddNewToken ||
+          route.name === ScreensEnum.AddNetwork
       ) ?? false;
 
-    if (isPopup && isCreateWalletScreensOpened && isReady) {
-      // clear previous navigation state and leave only ScreenEnum.ImportAccount route when click by extension icon
-      setStoredValue(PERSISTENCE_KEY, JSON.stringify({ ...initialState, routes: initialState?.routes.slice(0, 1) }));
-      openMaximiseScreen();
+    if (isPopup && IsOpenOnlyFullViewScreen && isReady) {
+      openFullViewPage();
+    }
+
+    const isOpenSomeOfCreateWalletScreens =
+      initialState?.routes.some(
+        route =>
+          route.name === ScreensEnum.CreateANewWallet ||
+          route.name === ScreensEnum.VerifyMnemonic ||
+          route.name === ScreensEnum.ImportWallet
+      ) ?? false;
+
+    if (isPopup && isOpenSomeOfCreateWalletScreens && isReady) {
+      const parentRoute = initialState?.routes.find(
+        route => route.name === ScreensEnum.ImportWallet || route.name === ScreensEnum.CreateANewWallet
+      )?.name;
+
+      setStoredValue(
+        PERSISTENCE_KEY,
+        JSON.stringify({
+          ...initialState,
+          routes: [
+            { key: 'Welcome-YOKi4uV67S17eluroXFWt', name: 'Welcome' },
+            { key: `${parentRoute}-h-XhNwryARoSQpUS0E3f-`, name: parentRoute }
+          ]
+        })
+      );
+
+      openFullViewPage();
```

Comment:
This hook looks complicated and non-explicit: hard conditions, auto-generated keys `YOKi4uV67S17eluroXFWt`, `h-XhNwryARoSQpUS0E3f`, changing store state, etc.  

Rewrite this hook and `redirectToFullViewPage` function.  
You could make some query params (for example `initialScreen: ScreensEnum`) and if that value defined - push that route into `storedInitialState` inside `usePersistedNavigationState` hook